src/services/ego.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2022 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import egoTokenUtils from '@icgc-argo/ego-token-utils';
+import fetch from 'node-fetch';
+import urlJoin from 'url-join';
+import { getAppConfig } from '../config';
+import Logger from '../logger';
+
+const logger = Logger('Ego');
+
+export type EgoApplicationCredential = {
+  clientId: string;
+  clientSecret: string;
+};
+
+export type ScoreAuthClient = {
+  getAuth: () => Promise<string>;
+};
+
+type EgoAccessToken = {
+  access_token: string;
+  token_type: string;
+  expires_in: number;
+  scope: string;
+  groups: string;
+};
+
+type EgoAccessTokenError = {
+  error: string;
+  error_description: string;
+};
+
+const getEgoPublicKey = async (): Promise<string> => {
+  const config = await getAppConfig();
+
+  if (config.auth.jwtKey) {
+    return config.auth.jwtKey;
+  }
+
+  if (config.auth.jwtKeyUrl) {
+    const response = await fetch(config.auth.jwtKeyUrl);
+    return await response.text();
+  }
+
+  return '';
+};
+
+const getApplicationJwt = async (
+  applicationCredentials: EgoApplicationCredential,
+): Promise<string> => {
+  const config = await getAppConfig();
+
+  const url = urlJoin(
+    config.auth.egoRootRest,
+    `/oauth/token?client_id=${applicationCredentials.clientId}&client_secret=${applicationCredentials.clientSecret}&grant_type=client_credentials`,
+  );
+
+  const response = await fetch(url, {
+    method: 'POST',
+    headers: {
+      'Content-type': 'application/json',
+    },
+  });
+
+  const authResponse = await response.json();
+
+  if (authResponse.error) {
+    throw new Error(
+      `Failed to authorize application: ${(authResponse as EgoAccessTokenError).error_description}`,
+    );
+  }
+
+  return (authResponse as EgoAccessToken).access_token;
+};
+
+export const createScoreAuthClient = async () => {
```

Comment:
I copied only the necessary pieces from https://github.com/icgc-argo/platform-api/blob/develop/src/services/ego/scoreAuthClient.ts and https://github.com/icgc-argo/platform-api/blob/develop/src/services/ego/index.ts for this

Reply:
for naming consistency, we aren't getting auth for Score here, just getting Ego auth in general. lets rename to `createAuthClient`